src/Workspaces/Core/Portable/Extensions/OperationExtensions.cs

```diff
@@ -91,6 +91,14 @@ public static ValueUsageInfo GetValueUsageInfo(this IOperation operation)
                         //
                         return ValueUsageInfo.Write;
 
+                    case IPropertySubpatternOperation _:
+                        // A declaration pattern within a property sub-pattern is a
+                        // write for the declared local.
+                        // For example, 'x' is defined and assigned the value from 'obj.Property' below:
+                        //      if (obj is { Property : int x })
+                        //
+                        return ValueUsageInfo.Write;
```

Comment:
great comment.  was confused at first, but now i get it.   

for completeness, is `Property` recorded as a read somewhere?

Reply:
Yes, the IOperation tree has an `IPropertyReferenceOperation` for the reference to Property on left of the colon. This switch statement defaults it to a read operation. Doing a FAR on definition of `P` below does show this as a read reference:

```csharp
class C
{
    public object P { get; }
    void M(C c)
    {
        var x = c is { P: int i };
    }
}
```

![image](https://user-images.githubusercontent.com/10605811/72647736-131a7000-392e-11ea-8300-7f33b7862e46.png)